### ISSUE TITLE
Padding mismatch from Tacotron2 pre-processing

### ISSUE THREAD

**Issue by SiddGururani** [jik876/hifi-gan#53]
Hello, I ran into similar trouble as #52 while trying to fine-tune the Universal_V1 checkpoint using ground-truth aligned TC2 outputs.

In my case, the fine-tuning breaks as soon as it reaches the validation phase. The error message is something along the lines of: 
```
Using a target size (torch.Size([1, 80, 132])) that is different to the input size (torch.Size([1, 80, 131])). This will likely lead to incorrect results due to broadcasting. Please ensure they have the same size.
  val_err_tot += F.l1_loss(y_mel, y_g_hat_mel).item()
```

I found that the padding performed in [this line](https://github.com/jik876/hifi-gan/blob/4769534d45265d52a904b850da5a622601885777/meldataset.py#L61) is different than the padding performed in [Tacotron2](https://github.com/NVIDIA/tacotron2/blob/185cd24e046cc1304b4f8e564734d2498c6e2e6f/stft.py#L87). 

I dug deeper and this does introduce a consistent difference of 1 frame between the mel-specs generated by the trained TC2 model and the hifi-gan loss mel-specs.

Technically I could simply edit the padding code in this repo and get the mels to align perfectly but I wonder if this would cause issues since the model is trained using differently padded mel spectrograms. What are your thoughts?

**Comment by jik876** [comment]
Please understand that we are a bit busy with other work.
The padding is a matter of how to consider the center. You can easily figure out what the center is.
Since it doesn't affect quality significantly, cutting spectrograms to match its lengths can be the simplest way.

**Comment by SiddGururani** [comment]
No worries! Yes, I realize that the padding issue is not a major problem since there will only be a few samples of mismatch.
For reference, I simply changed the TC2 mel spectrogram generation code to pad using the same scheme as the hifi-gan code. All spectrogram lengths match now, and should also align correctly

**Comment by ghost** [comment]
Dear @jik876 
Can you please provide the code for fixing this issue?

**Comment by v-nhandt21** [comment]
Yes, I have the same "difference of 1 frame between the mel-specs generated by the trained TC2 model and the hifi-gan loss mel-specs." as you, @SiddGururani 

In my case, I remove the first frame of all mel in batch, I don't know this explanation is correct or not: Tacotron2 is AR model so it needs a zero-frame first to generate. => taco has more than 1 frame

When calculating the difference between two mel: (mel from teacher force and audio) : the element differs mean = 10^2

And finally, I still wonder about mel_loss in fine tune of hifigan,  mel_taco_input  and  mel_hat generate from audio, I don't know it is a good idea or not when using L1 (because it is generated in a different way), should we use DTW in the future (w/o padding), I still experiment with this.

Hope for your comment my idol @jik876 
 
![image](https://user-images.githubusercontent.com/36092323/137009370-f9e87ea8-2374-4886-8470-0ac07d705834.png)
